pkg/sql/set_cluster_setting.go

```diff
@@ -617,6 +617,28 @@ func setVersionSetting(
 			return err
 		}
 		return db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+			// Run the version bump inside the upgrade as high priority, since
+			// lease manager ends up reading the version row (with high priority)
+			// inside the settings table when refreshing leases. On larger clusters
+			// or with a large number of descriptors it's possible for normal
+			// transactions to be starved  (#95227).
+			// This is safe because we expected this transaction only do the following:
```

Comment:
I think this is a me issue but I don't see how these items say using a high priority txn is safe? I also don't understand how it may be unsafe 😅